examples/src/main/java/com/hedera/hashgraph/sdk/examples/MirrorNodeContractQueriesExample.java

```diff
@@ -0,0 +1,176 @@
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("Mirror Node contract queries Example Start!");
+
+        /*
+         * Step 0:
+         * Create and configure the SDK Client.
+         */
+        Client client = ClientHelper.forName(HEDERA_NETWORK);
+        // All generated transactions will be paid by this account and signed by this key.
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+        // Attach logger to the SDK Client.
+        client.setLogger(new Logger(LogLevel.valueOf(SDK_LOG_LEVEL)));
+
+        var operatorPublicKey = OPERATOR_KEY.getPublicKey();
+
+        /*
+         * Step 1:
+         * Create the contract
+         */
+        System.out.println("Creating new file...");
+        TransactionResponse fileCreateTxResponse = new FileCreateTransaction()
+            // Use the same key as the operator to "own" this file.
+            .setKeys(operatorPublicKey)
+            .setContents(SMART_CONTRACT_BYTECODE)
+            // The default max fee of 1 Hbar is not enough to create a file (starts around ~1.1 Hbar).
+            .setMaxTransactionFee(Hbar.from(2))
+            .execute(client);
+
+        TransactionReceipt fileCreateTxReceipt = fileCreateTxResponse.getReceipt(client);
+        FileId newFileId = fileCreateTxReceipt.fileId;
+
+        Objects.requireNonNull(newFileId);
+        System.out.println("Created new file with ID: " + newFileId);
+
+        var response = new ContractCreateTransaction()
+            .setAdminKey(client.getOperatorPublicKey())
+            .setGas(200000)
+            .setConstructorParameters(new ContractFunctionParameters().addString("Hello from Hedera."))
+            .setBytecodeFileId(newFileId)
+            .setContractMemo("Simple contract with string field")
+            .execute(client);
+
+        var contractId = Objects.requireNonNull(response.getReceipt(client).contractId);
+
+        /*
+         * Step 3:
+         * Wait for mirror node to import data
+         */
+        Thread.sleep(4000);
+
+        /*
+         * Step 4:
+         * Estimate the gas needed
+         */
+        var gas = new MirrorNodeContractEstimateGasQuery()
+            .setContractId(contractId)
+            .setFunction("getMessage")
+            .execute(client);
+
+        System.out.println("Gas needed for this query: " + gas);
+
+        /*
+         * Step 5:
+         * Do the query against the consensus node using the estimated gas
+         */
+        var callQuery = new ContractCallQuery()
+            .setContractId(contractId)
+            .setGas(gas)
+            .setFunction("getMessage")
+            .setQueryPayment(new Hbar(1));
+
+        var result = callQuery
+            .execute(client);
+
+        /*
+         * Step 6:
+         * Simulate the transaction for free, using the mirror node
+         */
+        var simulationResult = new MirrorNodeContractCallQuery()
```

Comment:
Would be good to expos all potential params. For instance i should be able to specify `gasLimit` etc here

Reply:
Added more params to the example